### DISCUSSION TITLE
Nextcloud get's stuck after System Reboot

### DISCUSSION THREAD

**Discussion by simon-77** [nextcloud/all-in-one#4648] [Questions] [answered]
### Steps to reproduce
1. Setup Nextcloud AIO (v8.2.1) via docker compose
2. When nextcloud is running, restart the Host OS

### Expected behavior
Nextcloud AIO mastercontainer will take care of starting all other containers

### Actual behavior
Most Nextcloud containers are up and running, except `nextcloud` container.
As `apache` container is in _starting_ phase waiting for `nextcloud` container, the Nextcloud AIO management interface is just showing the status of containers without the option to stop / start or restart the containers.

Even after ~ 10 minutes it still got stuck in this state.

### Workaround to get everything going again
A manual intervention:
- stop the `nextcloud-aio-apache` container manually
- via AIO web interface: stop all containers
- via AIO web interface: start all containers

Unfortunatelly it is not possible somehow intervene in the AIO web interface while the containers are starting, therefore one must manually stop the nextcloud-apache container.

### Host OS
Rockstor, Version: 4.6.1-0
(Rockstor built on openSUSE Leap 15.4)

#### Nextcloud AIO version
v8.2.1

#### Current channel
_latest_ channel

#### Other valuable info
Nextcloud AIO interface after system reboot (and waiting ~ 5 minutes):
![Screenshot_20240512_112331](https://github.com/nextcloud/all-in-one/assets/30923077/8684f9e3-423e-4bf7-a502-12b262f258cb)

docker container status:
![Screenshot_20240512_112824](https://github.com/nextcloud/all-in-one/assets/30923077/57bafd1c-94c0-476f-bc38-9af1a6a601c8)

Container Log Files:
[_nextcloud-aio-apache_logs.txt](https://github.com/nextcloud/all-in-one/files/15285731/_nextcloud-aio-apache_logs.txt)
[_nextcloud-aio-mastercontainer_logs.txt](https://github.com/nextcloud/all-in-one/files/15285732/_nextcloud-aio-mastercontainer_logs.txt)
[_nextcloud-aio-nextcloud_logs.txt](https://github.com/nextcloud/all-in-one/files/15285733/_nextcloud-aio-nextcloud_logs.txt)

**Comment by szaimen** [comment]
Hi, which docker version are you running? Can you post the output of `sudo docker info` here?

Also, can you check if the following makes it work?
1. Stop your containers from the AIO interface
2. Enable ipv6 in your docker daemon if not already done by following https://github.com/nextcloud/all-in-one/blob/main/docker-ipv6-support.md
3. After the docker daemon started correctly run `sudo docker network disconnect nextcloud-aio nextcloud-aio-mastercontainer` to disconnect the mastercontainer from the nextcloud-aio network.
4. now remove the nextcoud-aio network by running `sudo docker network rm nextcloud-aio`
5. Then run `sudo docker network create --subnet="fd12:3456:789a:2::/64" --driver bridge --ipv6 nextcloud-aio` in order to recreate the nextcloud-aio network
6. Finally run `sudo docker network connect nextcloud-aio nextcloud-aio-mastercontainer` to connect the mastercontainer to the network again.
7. At last, open the AIO interface and start your containers again

**Comment by simon-77** [comment]
Thank you for the detailed reply.

Here is the output of `docker info`:
```
Client:
 Version:    24.0.7-ce
 Context:    default
 Debug Mode: false
 Plugins:
  compose: Docker Compose (Docker Inc.)
    Version:  v2.27.0
    Path:     /home/admin/.docker/cli-plugins/docker-compose

Server:
 Containers: 34
  Running: 32
  Paused: 0
  Stopped: 2
 Images: 41
 Server Version: 24.0.7-ce
 Storage Driver: overlay2
  Backing Filesystem: btrfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 oci runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8e4b0bde866788eec76735cc77c4720144248fb7
 runc version: v1.1.10-0-g18a0cb0f32bc
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: builtin
 Kernel Version: 5.14.21-150400.24.100-default
 Operating System: Rockstor built on openSUSE Leap 15.4
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.52GiB
 Name: Kolibri
 ID: e110c039-df47-4f26-afbf-d51f7d686d43
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

I have executed the steps and successfully enabled ipv6 for this subnet:
```
admin@Kolibri:~> docker network inspect nextcloud-aio | grep EnableIPv6
        "EnableIPv6": true,

```

Afterwards I started the Nextcloud containers again and then rebooted my system - unfortunately the same issue persists:
![image](https://github.com/nextcloud/all-in-one/assets/30923077/5d800229-2167-468e-9162-4ba7b2f0a6ee)

**Comment by simon-77** [comment]
By the way:

Upgraded to *Nextcloud AIO v9.0.1* and the issue still persists unchanged ...

**Comment by szaimen** [comment, marked as answer]
Hi, I fear I cannot reproduce your issue. Everything works fine on my test instance even after a reboot. I suspect that something on your host is causing this which AIO is not in control over. So for example an external drive that you use as datadir. If that is not correctly mounted before docker starts, it might fail starting the container. Probably best to ask in a docker forum about this.

**Reply by simon-77** [reply]
> So for example an external drive that you use as datadir. If that is not correctly mounted before docker starts, it might fail starting the container.

That was an excellent hint, thank you.

I (unknowingly) used a mount path that was not defined in `/etc/fstab` but was mounted by some script of my Host OS (Rockstor).

I created a new `/etc/fstab` entry for my docker volume and used this as nextcloud datadir, now everything works as expected after reboot.

Thank you for pointing me into the right direction,
Cheers Simon
